haystack/nodes/prompt/prompt_node.py

```diff
@@ -97,24 +93,19 @@ def __init__(
             },
         )
         super().__init__()
-        self.prompt_templates: Dict[str, PromptTemplate] = {pt.name: pt for pt in get_predefined_prompt_templates()}  # type: ignore
-        self.default_prompt_template: Union[str, PromptTemplate, None] = default_prompt_template
+        self._prompt_templates_cache: Dict[str, PromptTemplate] = {}
+
+        # self._default_template needs to be initialized here because the setter self.default_prompt_template uses
+        # self.get_prompt_template(), which in turn checks is there is any default already set.
```

Comment:
```suggestion
        # If we don't set _default_template here Pylint fails with error W0201 because it can't see that
        # it's set in default_prompt_template, so we set it explicitly to None to avoid it failing
```